### ISSUE TITLE
Question about Mastodon nomenclature

### ISSUE THREAD

**Issue by scripting** [scripting/Scripting-News#241]
Comment re this [Scripting News. post](http://scripting.com/2022/11/23/141702.html).

**Comment by doenietzomoeilijk** [comment]
Either `social.` or `m.`, I think. Friends is a misnomer, because his friends don't do anything there, microblogs is a bit long, and anything involving the actual software name is, IMO, not fun if you ever decide to switch to different software. We also don't host our websites at nginx.domain.whatever Either, right?

**Comment by thejeshgn** [comment]
I am going with social for now.

**Comment by josephzitt** [comment]
I would avoid going with "masto" or things like it.  Since Mastodon is just one ActivityPub client, it might be like using "googlereader" as a name for RSS things (though without the silo-ness).

**Comment by scripting** [comment]
Thanks for the comments -- @doenietzomoeilijk, @thejeshgn, @josephzitt

First I'm not saying I'm going with anything in particular, but in favor of "masto" --

1. Remember how everyone used the term RSS to apply not just to the format but also to the reader software? I think that's what has happened here. Mastodon is the name of the whole thing, not just a product. It includes ActivityPub. No one but people like us know hat AP is. 

2. If you deploy a Mastodon server, you aren't going to replace it with some other brand of AP server, because if you did that all the links into the Masto server would break. Once you've put it up you're stuck with it. Eventually of course the links *will* break, but you aren't going to want to put another brand of AP server at the same address.

**Comment by harrego** [comment]
> If you deploy a Mastodon server, you aren't going to replace it with some other brand of AP server, because if you did that all the links into the Masto server would break.

I don't think it fits "in the spirit" of AP. Technically if another AP service wins out against Mastodon for whatever reason there is nothing that could stop some migration tool from providing a seamless experience for the user and third-party instances. Services like [Pleroma](https://pleroma.social/) do replicate the Mastodon experience while sitting on AP, there could be valid reason for an entire Masto instance to switch in future.

Anecdotally I have primarily seen the `social` subdomain used the most which judging from history could be the reason it wins out. Personally I like `activitypub` and related for the reason `ftp` makes sense, but it's also a little long for practical use, if only ActivityPub had a unique acronym ;-)

**Comment by doenietzomoeilijk** [comment]
I kinda disagree on point 1 here; yes, lots of reader software had RSS (or "feed"!) in their names, but RSS is what I'd like to call a core tech. And before Atom was around, RSS *was* the tech. In this current situation, AP is the tech, Masto is just one implementation out of many. I understand what you're saying when you say "no one but people like us", but doesn't that also imply that we have a bit of a responsibility to educate other people? Just like we tell all our family members and friends that no, it's not smart to use the same password for everything, especially if it's your birthday?

As for point 2... yeah, I suppose that's somewhat true (although you might go a long way with redirects).

That being said... using a "brand name" for a "technology thing" (especially if it might actually not be that brand!) rubs me the wrong way. It'd be like naming my mailserver "gmail.whatever.domain" even when it's running Postfix, with the excuse that gmail is a known name where postfix probably isn't.

**Comment by scripting** [comment]
### A Hello World for ActivityPub?

Since there are all these ActivityPub experts here, may I ask a simple question as an AP newbie...

How do I write a Hello World for ActivityPub?

I could write a Hello World HTML page in a minute or less. Same with an RSS feed with one item that said Hello World.

Serious question, can you show me a Hello World for AP? I want to get started but every time I try to read about it I keep wondering why don't they help me see the forest before walking me through all the freaking trees? (AWS docs are like that too, they drive me crazy.)

**Comment by berkes** [comment]
I've gone for "say" as subdomain. say.example.com. It's a trial of gotosocial, which is much lighter and far easier than Mastodon. But also unfinished.
